cmd/main.go

```diff
@@ -122,6 +126,18 @@ func runOperator() {
 	}
 }
 
+// watchForConfigChanges keeps checking if the informed configuration has changed, and in this case
+// makes the operator exit. The new configuration should take place upon new instance started.
+func watchForConfigChanges(mgr *config.Manager) {
+	for {
+		if mgr.ConfigHasChanged() {
```

Comment:
this is fine for now for prototype testing, but add a TODO comment that when the operator changes are in place, we quit watching and reacting to updates in this driver, and will defer to the operator to watch for config changes, and restart our entire daemonset when it sees them, as I noted in my comment https://github.com/openshift/csi-driver-shared-resource/pull/60#pullrequestreview-779831515 where we follow / employ the library-go related utilities like OCM-O / OCM

If there was something already analogous to this in OCM, I'm missing it.  Do you have such a pointer?

Reply:
Hmmm ... as I read the acceptance criteria in BUILD-340, it says "The shared resource CSI driver **operator** watches for create/update on this ConfigMap and processes config changes dynamically, or considers restart if appropriate."

Now, if you want to work with @adambkaplan to break up BUILD-340, then sure let's discuss that in our team meetings.

But as I read that acceptance criteria, your watching for config changes needs to move out of here and into the opeator.

Reply:
OK I don't remember this precise element coming up in those prior discussions, about deferring on making a change in the operator.  I remember discussion on the mechanics on how to do it.

Other than trying again to sort out those details, and deciding what is done now vs. later, I don't have any other suggestions.